net/pfSense-pkg-pfBlockerNG-devel/files/usr/local/pkg/pfblockerng/pfblockerng.inc

```diff
@@ -1055,7 +1058,8 @@ function pfb_create_dnsbl_cert() {
 
 	$pkey	= openssl_pkey_new($pkey_config);
 	$csr	= openssl_csr_new($dn, $pkey);
-	$cert	= openssl_csr_sign($csr, NULL, $pkey, 3650);
+	$cert	= openssl_csr_sign($csr, NULL, $pkey, 3650, array('digest_alg'=>'sha256',
+				'x509_extensions' => 'server'), cert_generate_serial());
```

Comment:
You probably also want to set `"req_extensions" => "req_server"` in this. Or, even better, use the functions we already have like `cert_create`/`csr_generate`/`csr_sign` and so on rather than using OpenSSL functions directly here.

Is there something you are doing which isn't supported by the built-in functions? If you need self-signed, you can still follow the example set by the GUI cert in `system.inc` inside `system_webgui_create_certificate()` and do something like this:

    cert_create($cert, null, 2048, 389, $dn, "self-signed", "sha256");

Note that you shouldn't use 3650 for server cert lifetimes anymore. The current max is 825 and this fall it will be lowered to 398. We've already updated 2.5.0 to recommend 389. See https://redmine.pfsense.org/issues/9825 for details.

Reply:
Thanks, new commit pushed that uses cert_create as suggested.